### ISSUE TITLE
"Run jest" file context menu item

### ISSUE THREAD

**Issue by hornta** [jest-community/vscode-jest#509]
I would like a right click on file option "Run jest"somewhere in this menu.

![image](https://user-images.githubusercontent.com/4553604/66686707-9ae24580-ec80-11e9-9ffd-2817eb9948a9.png)

**Comment by connectdotz** [comment]
in a normal situation, jest is run in watch mode, i.e. it will run your tests automatically (when something related changed). Can you elaborate on the use case that you prefer to trigger it manually?

**Comment by benasher44** [comment]
FWIW, this is probably the main feature keeping us from switching away from https://github.com/firsttris/vscode-jest-runner

**Comment by connectdotz** [comment]
hi, the new v4 release does come with a context menu you can use to trigger the test run on-demand:

![Screen Shot 2021-06-13 at 9 39 45 PM](https://user-images.githubusercontent.com/891093/121829293-1be0b400-cc90-11eb-9395-40d838ac9267.png)

Here is the instruction on how to set up your env for the interactive mode: https://github.com/jest-community/vscode-jest#how-to-trigger-the-test-run (click expand the "interactive mode"). Let me know if you have any issues using it.

**Comment by benasher44** [comment]
Oh great, thanks! Will give that a try

**Comment by benasher44** [comment]
Do you think something like #288 is in the cards to better surface this (when file is wrapped in a describe)?

**Comment by connectdotz** [comment]
Yeah it's quite doable, we might do this as part of supporting the official vscode test explore, which hopefully will become available in a couple of months...
